### PR TITLE
Adds new sounds to the microwave

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
add: Microwaves have new sounds!
/:cl:

[Demo here.](https://puu.sh/yhabJ/f71dc95b0e.mp3)

Another kitchen PR. I realized that microwaves might sound a lot neater with looping sounds now that the framework is there, so I added them.

### CODE REVIEW

File: code/modules/food_and_drinks/kitchen_machinery/microwave.dm [hunk 1]
```diff
@@ -15,6 +15,7 @@
 	var/broken = 0 // ={0,1,2} How broken is it???
 	var/max_n_of_items = 10 // whatever fat fuck made this a global var needs to look at themselves in the mirror sometime
 	var/efficiency = 0
+	var/datum/looping_sound/microwave/beep_beep_beep
```
Comment (lzimann) [inline code/modules/food_and_drinks/kitchen_machinery/microwave.dm]: name this variable something else that makes sense

**Comment by ninjanomnom** [comment]
Sounds too fast imo and part of the point of this looping datum is to allow different sounds mid loop and avoid hearing the point where it loops.

**Comment by Xhuis** [comment]
I originally had three sounds, but I couldn't find a good point that made it loop seamlessly; 3 has a pause and 1 overlaps far too much. I also can't put it as the actual duration of the sound because there's a very slightly delay, I guess because of Byond.

**Comment by joctopus** [comment]
AquaStation where

File: code/modules/food_and_drinks/kitchen_machinery/microwave.dm (hunk 1 above)
Comment (Qustinnus) [reply code/modules/food_and_drinks/kitchen_machinery/microwave.dm]: just call it soundloop, it's what literaly every piece of code does with soundloops.

**Comment by Buggy123** [comment]
If the sounds aren't great, maybe try recording some? I mean, odds are that you almost _certainly_ have a phone and a microwave, so maybe try recording it and see if it's any good.

And hey, if you do, your work will forever be immortalized as part of SS13. Any time someone uses a microwave, they'll think of you. That's a **warming** thought, isn't it?

**Comment by Xhuis** [comment]
I do have a recording of a microwave running a 30-second timer that the sounds are taken from, but I had no luck getting them to work well in loop.

**Comment by DoctoreMad** [comment]
Oh boy, remember how your microwave has BEEPS for the OLD MEN?
Now your space station microwave is into it too!

File: code/datums/looping_sounds/machinery_sounds.dm [hunk 2]
```diff
@@ -21,4 +21,14 @@
...
+	volume = 40
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/datum/looping_sound/microwave
+	start_sound = 'sound/machines/microwave/microwave_on.ogg'
+	start_length = 8
+	mid_sounds = list('sound/machines/microwave/microwave_hum.ogg' = 1)
+	mid_length = 2
+	end_sound = 'sound/machines/microwave/microwave_BEEEP.ogg'
```
Comment (nicbn) [inline code/datums/looping_sounds/machinery_sounds.dm]: No capital letters at resource file names please

**Comment by Xhuis** [comment]
@ninjanomnom has provided some new sounds; I've updated the OP with an .mp3 demo taken from in-game.

**Comment by Qustinnus** [comment]
the delay seems to be a bit too short, its cutting to the next soundclip rather oddly

**Comment by ninjanomnom** [comment]
it was kinda the point, a lot of holder microwaves have a sound their disc makes as it leans from side to side

**Comment by Qustinnus** [comment]
must be an american thing :^)

**Comment by ninjanomnom** [comment]
\>the very concept of degradation and the second law of thermodynamics is an american thing

**Comment by Qustinnus** [comment]
haha funny american

**Comment by Buggy123** [comment]
As a American, I can confirm that degradation is absolutely an American thing. Just look at how much of our stuff is falling apart!

**Review by lzimann — CHANGES_REQUESTED** [review]
gotta change the var's name

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**